engine/render/src/render/program_utils.cpp

```diff
@@ -82,19 +93,36 @@ namespace dmRender
 
             assert(name_str_length > 0);
 
-            // For uniform arrays, OpenGL returns the name as "uniform[0]",
-            // but we want to identify it as the base name instead.
-            for (int j = 0; j < name_str_length; ++j)
+            if (program_language_glsl)
             {
-                if (buffer[j] == '[')
+                // For uniform arrays, OpenGL returns the name as "uniform[0]",
+                // but we want to identify it as the base name instead.
```

Comment:
Is this something we want handled by the content pipeline?

And, alternatively, we can implement a dmGraphics::GetUniformNameHash()

Reply:
Yes that would be ideal. Right now I don't know that we can generate proper reflection data for all opengl programs, I think I need to do some research.. I would assume SPIR-V would have something for this, but I'm not sure.